overlord/devicestate/handlers_serial.go

```diff
@@ -520,8 +522,7 @@ func getSerial(t *state.Task, regCtx registrationContext, privKey asserts.Privat
 	}
 
 	if !shouldRequest {
-		t.Logf("skipping getting serial, store is marked as offline")
-		return nil, nil, nil
```

Comment:
intersting that we didn't spot this, anyway in general one needs to be extra careful when returning nils without errors, it's a  contract that at most is viable for very close functions, which here is the case, but the caller was still unprepared and the situation untested